app/scripts/views/mixins/password-mixin.js

```diff
@@ -45,6 +46,29 @@ define(function (require, exports, module) {
         // IE8 blows up when changing the type of the input field. Other
         // browsers might too. Ignore the error.
       }
+    },
+
+    onPasswordKeypress: function (event) {
+      var val = this.getElementValue('.password').length;
+      if (val > 6) {
+        this.setPasswordWarning(true);
+      } else {
+        this.setPasswordWarning(false);
+      }
+    },
+
+    setPasswordWarning: function (isVisible) {
+      try {
+        var passwordhelp = this.$('.input-help');
+        if (isVisible) {
```

Comment:
If `isVisible` is true, this will actually hide the password help. Should be `!isVisible`.